### ISSUE TITLE
(new-method) add VICReg implementation

### ISSUE THREAD

**Issue by SauravMaheshkar** [SauravMaheshkar/Self-Supervised-Learning#8]
Add Implementation of [VICReg: Variance-Invariance-Covariance Regularization for Self-Supervised Learning)](https://arxiv.org/abs/2105.04906).

We mostly depend on the official code released on [facebookresearch/vicreg](https://github.com/facebookresearch/vicreg). 

<details>
<summary><b>Citation</b></summary>

```
@inproceedings{bardes2022vicreg,
  author  = {Adrien Bardes and Jean Ponce and Yann LeCun},
  title   = {VICReg: Variance-Invariance-Covariance Regularization For Self-Supervised Learning},
  booktitle = {ICLR},
  year    = {2022},
}
```

</details>

**Notebooks:-**

- [x] Add PyTorch Implementation (ported from the original implementation) ([344ba0b](https://github.com/SauravMaheshkar/Self-Supervised-Learning/pull/9/commits/344ba0b1da24fd9e39b76f4b119eb01a6b21eb96))
- [x] Add Tensorflow Implementation ([eed7d38](https://github.com/SauravMaheshkar/Self-Supervised-Learning/pull/9/commits/eed7d385ae93495b28e0118618541b78796dbad8))
- [ ] Add JAX Implementation (`Flax`)

**Modules:-**

- [x] PyTorch
    - [x] augmentations: add `Gaussian Blur` and `Solarization` implementations with related tests. ([8d9d770](https://github.com/SauravMaheshkar/Self-Supervised-Learning/pull/9/commits/8d9d770cb851f8725efbc5a38f6c6a6036763a54))
    - [x] optimizers: add `LARS` implementation with related tests. (d0dd63dcf608617f3cbd4d76ec342f1832f8b319)
    - [x] model: add `VICReg` implementation with related tests. (dca4ce233d9bbac3768696ae90566d8c9d00eae5)
- [x] Tensorflow
    - [x] augmentations: add `shuffle__zipped_outputs` with arbitrary # of inputs, `scale_image`, `gaussian_blur`,  `color_jitter`, `solarize`, `color_drop` and `random_apply` with related tests. (8e5ae66ce2c1ed36f61881a4918451db8853066c)
    - [x] model: add `VICReg` implementation with related tests. (daee4368ab7d10655f056cc720c45ed0f1ac2107)

**Comment by noahweber1** [comment]
@SauravMaheshkar this is great effort Saurav, but help me understand. How does this connect to dna-diffusion.

I mean that we will need to regularize is clear.

But we need to implement this directly in either UNET or VQ_VAE notebooks and then test wether it performs as it should perform there. Meaning direct evaluation?

**Comment by noahweber1** [comment]
no worries
